docs/source/pytest.rst

```diff
@@ -0,0 +1,354 @@
+        entry_points={
+            "pytest11": [
+                "cocotb = cocotb_tools.pytest.plugin",
+            ],
+        },
+    )
+
+By defining the global variable ``pytest_plugins`` when using a ``conftest.py`` file
+(which must be located in the root of the project):
+
+.. code:: python
+
+    pytest_plugins = ("cocotb_tools.pytest.plugin",)
+
+By defining the ``PYTEST_PLUGINS`` environment variable:
+
+.. code:: shell
+
+    export PYTEST_PLUGINS="cocotb_tools.pytest.plugin"
+
+By using the ``-p <plugin>`` option when invoking the `pytest`_ command line interface:
+
+.. code:: shell
+
+    pytest -p cocotb_tools.pytest.plugin ...
+
+Building and Testing
+====================
+
+:py:class:`cocotb_tools.pytest.hdl.HDL` wraps :py:class:`cocotb_tools.runner.Runner`
+allowing to fully configure the cocotb runner by using the command line arguments ``--cocotb-*``,
+configuration files like `pyproject.toml`_ or `fixture`_ arguments.
+
+The plugin provides a ``hdl`` fixture that will create a new instance of :py:class:`cocotb_tools.pytest.hdl.HDL`
+with the `pytest`_ built-in `request`_ fixture that is providing information of the requesting test function
+including the current configuration of `pytest`_.
+
+Example content of a ``conftest.py`` file:
+
+.. code:: python
+
+    import pytest
+    from cocotb_tools.pytest.hdl import HDL
+
+
+    @pytest.fixture(name="sample_module")
+    def sample_module_fixture(hdl: HDL) -> HDL:
+        """Define HDL design by adding HDL source files to it.
+
+        Args:
+            hdl: Fixture created by the cocotb pytest plugin, representing a HDL design.
+
+        Returns:
+            Representation of HDL design with added HDL source files.
+        """
+        hdl.sources = (
+            # List HDL source files,
+            "sample_module.sv",
+        )
+
+        return hdl
+
+
+Example content of the ``test_sample_module.py`` file:
+
+.. code:: python
+
+    import pytest
+    from cocotb_tools.pytest.hdl import HDL
+
+
+    # Without providing positional arguments to the cocotb decorator,
+    # the plugin will use the current file as the cocotb testbench (a Python file with cocotb tests).
+    # If the 'toplevel' option was not provided, it will be derived from the name of the first positional argument
+    # but with a removed 'test_*' prefix or '*_test' suffix.
+    @pytest.mark.cocotb  # equivalent to @pytest.mark.cocotb("test_dut", toplevel="dut")
+    def test_sample_module(sample_module: HDL) -> None:
+        """Build HDL design and run HDL simulator to execute cocotb tests.
+
+        Args:
+            sample_module: An instance of a defined HDL design.
+        """
+        sample_module.test()
+
+
+    # A @pytest.mark.cocotb or @cocotb.test decorator is not required if the test function
+    # starts with a 'test_*' prefix, is a coroutine function (``async``) and has a ``dut`` argument.
+    async def test_some_dut_feature(dut) -> None:
+        """cocotb test for DUT."""
```

Comment:
I had several questions here that were answered below.
Remove this and either have nothing or a sentence explaining the examples below use this fixture.
```suggestion

```